src/main/java/dev/emortal/minestom/minesweeper/view/InteractionManager.java

```diff
@@ -118,12 +121,16 @@ private void onClick(@NotNull PlayerBlockInteractEvent event) {
             return;
         }
 
+        // If the block is already carpet (a flag), we want to remove the flag
         if (this.board.isFlagged(x, z)) {
-            // If the block is already carpet (a flag), we want to remove the flag
-            player.playSound(Sound.sound(SoundEvent.ENTITY_ITEM_PICKUP, Sound.Source.MASTER, 0.6F, 1.8F));
-            this.actionBar.decrementFlags();
+            if (!(this.board.getFlag(x, z) == Block.BLACK_CARPET)
```

Comment:
Believe it should be the following:
```suggestion
            if (this.board.getFlag(x, z) != Block.BLACK_CARPET
```